src/Microsoft.AspNetCore.Session/DistributedSession.cs

```diff
@@ -191,28 +218,52 @@ private void Load()
         {
             if (_isModified)
             {
-                var data = await _cache.GetAsync(_sessionKey);
-                if (_logger.IsEnabled(LogLevel.Information) && data == null)
+                if (_logger.IsEnabled(LogLevel.Information))
                 {
-                    _logger.SessionStarted(_sessionKey, Id);
+                    try
+                    {
+                        var data = await _cache.GetAsync(_sessionKey);
+                        if (data == null)
+                        {
+                            _logger.SessionStarted(_sessionKey, Id);
+                        }
+                    }
+                    catch (Exception exception)
+                    {
+                        _logger.SessionCacheReadException(_sessionKey, exception);
+                        return;
```

Comment:
That works but you get two exceptions instead of one logged. I guess the duplication there isn't a problem.